core/src/epicli/data/common/ansible/playbooks/roles/upgrade/tasks/kubernetes/reconfigure-auth-service-app.yml

```
@@ -1,47 +1,56 @@
 ---
-- name: Change keycloak stateful set to use {{ image_registry_address }}
+- name: "Change keycloak stateful set to use {{ image_registry_address }}"
+  run_once: true
+  environment:
+    KUBECONFIG: /home/{{ admin_user.name }}/.kube/config
   block:
     - name: upgrade-master | Get keycloak statefulset name
-      environment:
-        KUBECONFIG: /home/{{ admin_user.name }}/.kube/config
       shell: |-
-        kubectl get statefulsets.apps --all-namespaces -o=jsonpath='{range .items[*]}{.metadata.name}{"\t"}{.spec.template.spec.containers[].image}{"\n"}{end}'|
+        kubectl get statefulsets.apps --all-namespaces -o=jsonpath={{ _jsonpath }} |
         grep -i keycloak |
         awk '{print $1}'
+      vars:
+        _jsonpath: >-
+          '{range .items[*]}{.metadata.name}{"\t"}{.spec.template.spec.containers[].image}{"\n"}{end}'
```

```suggestion
          '{range .items[*]}{.metadata.name}{"\t"}{.spec.template.spec.containers[].image}{"\t"}{.metadata.namespace}{"\n"}{end}'
```
We could get all three properties in one query, then the next two task could be removed.